workers/generic-worker/fileutil/fileutil.go

```diff
@@ -28,8 +29,14 @@ func WriteToFileAsJSON(obj any, filename string) error {
 	} else {
 		log.Printf("Saving file %v", filename)
 	}
-
-	return os.WriteFile(filename, append(jsonBytes, '\n'), 0644)
+	tempFilename := filename + "-" + slugid.Nice()
+	if err := os.WriteFile(tempFilename, append(jsonBytes, '\n'), 0644); err != nil {
```

Comment:
what is the reason behind writing to a temp file first and then moving it?

Reply:
So the cache mapping file doesn't get corrupted if the worker gets interrupted mid-write.

Reply:
Note, this could also be the cause of https://mozilla.sentry.io/issues/3332121580/?project=6462337

```
*json.SyntaxError

invalid character '\x00' looking for beginning of value
```